lib/govuk_ab_testing/experiment.rb

```diff
@@ -0,0 +1,36 @@
+module GovukAbTesting
+  class Experiment
+    attr_reader :ab_test, :request
+
+    delegate :request_header, :cookie_name, :response_header, to: :ab_test
+
+    # @param experiment_name [String] Lowercase experiment name, like `example`
+    # @param request [ApplicationController::Request] the `request` in the
+    # controller.
+    def initialize(experiment_name, request)
```

Comment:
The way I'm expecting to use this is to have one "global" `AbTest` object (probably sitting in my `application_controller`), and then use _that_ to spawn these `Experiment`s. For example:

```
# application_controller.rb
@ab_test = AbTest.new("educationnavigation")

# some_extended_controller.rb
@requested_variant = @ab_test.variant_from request
```

Of course, you can have this "raw" constructor, as well. I just think it's weird to keep reusing the magic string that is your `experiment_name`, when you could just stick it in your removed concept of the `AbTest`, and essentially use that object as a factory.

At the very least I think it would be nice to have a constructor that accepts the pre-existing `AbTest` as an argument (again to avoid reuse of magic strings).